src/integer_mod_q/mat_zq/arithmetic/mul_scalar.rs

```diff
@@ -0,0 +1,362 @@
+    }
+}
+
+#[cfg(test)]
+mod test_mul_zq {
+
+    use crate::integer_mod_q::{MatZq, Zq};
+    use std::str::FromStr;
+
+    /// Checks if matrix multiplication works fine for both borrowed
+    #[test]
+    fn borrowed_correctness() {
+        let mat1 = MatZq::from_str("[[42, 17],[8, 6]] mod 61").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = MatZq::from_str("[[84, 34],[16, 12]] mod 61").unwrap();
+        let integer = Zq::try_from((2, 61)).unwrap();
+
+        let mat1 = &mat1 * &integer;
+        let mat2 = &integer * &mat2;
+
+        assert_eq!(mat3, mat1);
+        assert_eq!(mat3, mat2);
+    }
+
+    /// Checks if scalar multiplication works fine for both owned
+    #[test]
+    fn owned_correctness() {
+        let mat1 = MatZq::from_str("[[42, 17],[8, 6]] mod 61").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = MatZq::from_str("[[84, 34],[16, 12]] mod 61").unwrap();
+        let integer1 = Zq::try_from((2, 61)).unwrap();
+        let integer2 = Zq::try_from((2, 61)).unwrap();
+
+        let mat1 = mat1 * integer1;
+        let mat2 = integer2 * mat2;
+
+        assert_eq!(mat3, mat1);
+        assert_eq!(mat3, mat2);
+    }
+
+    /// Checks if scalar multiplication works fine for half owned/borrowed
+    #[test]
+    fn half_correctness() {
+        let mat1 = MatZq::from_str("[[42, 17],[8, 6]] mod 61").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = mat1.clone();
+        let mat4 = mat1.clone();
+        let mat5 = MatZq::from_str("[[84, 34],[16, 12]] mod 61").unwrap();
+        let integer1 = Zq::try_from((2, 61)).unwrap();
+        let integer2 = Zq::try_from((2, 61)).unwrap();
+
+        let mat1 = mat1 * &integer1;
+        let mat2 = &integer2 * mat2;
+        let mat3 = &mat3 * integer1;
+        let mat4 = integer2 * &mat4;
+
+        assert_eq!(mat5, mat1);
+        assert_eq!(mat5, mat2);
+        assert_eq!(mat5, mat3);
+        assert_eq!(mat5, mat4);
+    }
+
+    /// Checks if scalar multiplication works fine for matrices of different dimensions
+    #[test]
+    fn different_dimensions_correctness() {
+        let mat1 = MatZq::from_str("[[42],[0],[2]] mod 61").unwrap();
+        let mat2 = MatZq::from_str("[[2, 6, 5],[4, 42, 3]] mod 61").unwrap();
+        let mat3 = MatZq::from_str("[[84],[0],[4]] mod 61").unwrap();
+        let mat4 = MatZq::from_str("[[4, 12, 10],[8, 84, 6]] mod 61").unwrap();
+        let integer = Zq::try_from((2, 61)).unwrap();
+
+        assert_eq!(mat3, &integer * mat1);
+        assert_eq!(mat4, integer * mat2);
+    }
+
+    /// Checks if matrix multiplication works fine for large values
+    #[test]
+    fn large_entries() {
+        let mat1 = MatZq::from_str(&format!("[[1],[{}],[4]] mod {}", i64::MAX, u64::MAX)).unwrap();
+        let mat2 = MatZq::from_str(&format!("[[3]] mod {}", u64::MAX)).unwrap();
+        let mat3 = MatZq::from_str(&format!(
+            "[[3],[{}],[12]] mod {}",
+            3 * i64::MAX as i128,
```

Comment:
You missed at least one spot to apply this to. If you do this change, do it everywhere. It is still optional, but it should be consistent for the test cases.